### ISSUE TITLE
a `check` in unittest generated by template doesn't show actual value

### ISSUE THREAD

**Issue by alehander92** [nim-lang/Nim#6736]
```nim
import unittest

type
  A = object
    b: int

template t: untyped =
  check(a.b == 2)

suite "0":
  test "1":

    var a = A(b: 0)
    check(a.b == 2)

suite "1":
  test "1":

    var a = A(b: 0)
    t()
```

[Suite] 0
    t2.nim(14, 14): Check failed: a.b == 2
    a.b was 0
  [FAILED] 1

[Suite] 1
    t2.nim(8, 12): Check failed: a.b == 2
  [FAILED] 1

**Comment by GULPF** [comment]
This should be fixed by #6446. For most use cases, I would recommend you to use the latest devel version of Nim. It can be installed easily with [choosenim](https://github.com/dom96/choosenim), or you can build it manually.

**Comment by chrisheller** [comment]
While I was looking at unittest.nim for #12031,  I took a quick look at this.  The reason that the template doesn't print the actual value is because the check macro is testing for nnkIdent for the equality test, but the template has not been resolved yet,  so the "==" is still nnkOpenSymChoice at that point. 

Updating unittest.nim with this change makes the above test code work properly.  

```diff
diff --git a/lib/pure/unittest.nim b/lib/pure/unittest.nim
index 469e75c16..0b65fc888 100644
--- a/lib/pure/unittest.nim
+++ b/lib/pure/unittest.nim
@@ -610,7 +610,7 @@ macro check*(conditions: untyped): untyped =
 
     var counter = 0
 
-    if exp[0].kind == nnkIdent and
+    if exp[0].kind in {nnkIdent, nnkOpenSymChoice} and
         $exp[0] in ["not", "in", "notin", "==", "<=",
                     ">=", "<", ">", "!=", "is", "isnot"]:
```

There is another place in that same section of code that also checks for nnkIdent,  but I didn't have an immediate test case to trigger any incorrect behavior there,  so I didn't change anything.
